com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -1800,8 +1804,10 @@ private List<TaskResult> runTasks(IProgress monitor) throws IOException {
                     }
                 }
                 TimeProfiler.stop();
-
-                boolean shouldRun = (!allOutputExists || !allSigsEquals) && !completedTasks.contains(task);
+                // game.project is always last task
+                boolean isLastTask = completedTasks.size() + 1 == buildTasks.size();
```

Comment:
It feels a bit unstable to rely on this. Sure, it is not likely to change, but still.... I would feel better if we did something like:

```
boolean isGameProjectTask = (task.getBuilder() instanceof GameProjectBuilder.class);
```

Reply:
If the last task isn't game.project - the build is broken. Like totally broken.

Reply:
Ok then! It might be worth adding a comment about why we do it like this. Or add a reference to this ticket.

Reply:
Ok. I'll do. 
Old logic before my refactoring had priority to use it only for game.project to make sure it builds the last one. Now when we have real build tree, it happens naturally (every task requires all the input builds to be done). 

Game.project has to be the last one, because it packs everything. If some task for some reason will be called after game.project, that means something goes really wrong and changes from such tasks will never be in the final bundle. I'll think how to add comment and somehow make sure the last task is game.project (maybe I'll add exception if it's not)

Reply:
Ok. I'll do. 
Old logic before my refactoring had priority to use it only for game.project to make sure it builds the last one. Now when we have real build tree, it happens naturally (every task requires all the input builds to be done). 

Game.project has to be the last one, because it packs everything. If some task for some reason will be called after game.project, that means something goes really wrong and changes from such tasks will never be in the final bundle. I'll think how to add comment and somehow make sure the last task is game.project (maybe I'll add exception if it's not)